presc/spatial_distribution/categorical_distance.py

```diff
@@ -199,113 +247,182 @@ def goodall_frequency(self, counts):
 
     def distance_to_data(self, dpoint, metric, distance_sample=0.013):
         """Computes an estimate of the average distance of a data point to every
-        other data points by taking a random sample of a given size and avereging
+        other data point by taking a random sample of a given size and avereging
         the result
         Args:
              dpoint(Pandas Series): Pandas Series data point to work with
              metric (str): similarity metric used to compute the distance
              distance_sample(numeric): Number that if from 0 to 1  indicates the percentage
                               from the total data that should be used to
                               estimate the average distance and if greater indicates the precise number
-                              of samples to be take"""
+                              of samples to be take
+        Returns(float): Estimate of average distance from a point to the data"""
 
-        if distance_sample < 1:
+        metric, type = self.get_metric(metric)
+        if type == "categorical":
+            data = self.cat_data
+            dpoint = dpoint[dpoint.apply(lambda x: isinstance(x, str))]
```

Comment:
To subset the categorical columns (and numerical columns below), it would be nicer to just select the corresponding elements of the datapoint vector directly. `dpoint` should be a Series indexed by the column names, so you should be able to do `dpoint[data.columns]`